cli/atlas/formatting.go

```diff
@@ -40,17 +41,23 @@ func ServerURL(str string) (string, error) {
 }
 
 // ProjectRoot determines the root directory of an application. The project
-// root is considered to be anything after go/src/...
-func ProjectRoot(dirString string) (string, error) {
-	dirs := strings.Split(dirString, "/")
-	for i, dir := range dirs {
-		if strings.ToLower(dir) == "go" && i+1 < len(dirs) {
-			if i+2 < len(dirs) && strings.ToLower(dirs[i+1]) == "src" {
-				return strings.Join(dirs[i+2:], "/"), nil
-			}
-		}
+// root is considered to be anything in $GOPATH/src or below
+func ProjectRoot(gopath, workdir string) (string, error) {
+	if gopath == "" {
+		return "", errMissingGOPATH
 	}
-	return "", errInvalidProjectRoot
+	if len(workdir) < len(gopath) {
```

Comment:
do you use the `gopath` for anything other than this validation? I'm wondering why we care so much if it's not valid.

A valid use case could be that someone bootstraps their app in some dumb file location then after that change their gopath to point to it. They'd be dumb, but who are we to judge?

An alternative would be to give them a warning if their location isn't in their gopath, instead of a hard fail.

Reply:
I'm relatively unopinionated on it, just providing an alternative perspective. I'll be happy with whichever choice you think is most useful :)

Reply:
IMO, people should have to initialize projects in their `GOPATH` because it's more in line with convential Go practices.